src/app/shared/operators/progressive-list-from-list.ts

```diff
@@ -0,0 +1,44 @@
+import { animationFrames, Observable } from 'rxjs';
+import { map, scan, take } from 'rxjs/operators';
+
+/**
+ * Generate stream providing a list populated by chunks at each interval
+ *
+ * IE: chunkSize = 2, list = [1, 2, 3, 4, 5, 6] streams following values:
+ * 1. [1, 2]             → merge 2 first items
+ * 2. [1, 2, 3, 4]       → merge 2 new items
+ * 3. [1, 2, 3, 4, 5, 6] → merge 2 new items
+ *
+ * IE: chunkSize = 2, startIndex = 3, list = [1, 2, 3, 4, 5, 6] streams following values:
+ * 1. [1, 2, 3]          → until startIndex
+ * 2. [1, 2, 3, 4, 5]    → merge 2 new items
+ * 3. [1, 2, 3, 4, 5, 6] → merge the last remaining item
+ *
+ * @param list array of any value
+ * @param options
+ * @returns Observable of list
+ */
+export function progressiveListFromList<T>(list: T[], options: Options = {}): Observable<T[]> {
```

Comment:
`progressiveListFromList` looks like it returns a list while it creates an observable.
I don't see how we can do `fromABC` as we are not in the Rx lib, it will never be very clear. So you could do something like `progressiveObservableFromList`

Reply:
Fixed